weldx/util/xarray.py

```diff
@@ -529,13 +530,47 @@ def xr_check_coords(dax: xr.DataArray, ref: dict) -> bool:
                 raise DimensionalityError(
                     units,
                     check["dimensionality"],
-                    f"\nDimensionality mismatch in coordinate '{key}'\n"
+                    extra_msg=f"\nDimensionality mismatch in coordinate '{key}'\n"
                     f"Coordinate has unit '{units}', expected '{dim}'",
                 )
 
     return True
 
 
+def xr_check_dimensionality(da: xr.DataArray, units_ref: Union[str, pint.Unit]):
+    """Check if a `DataArray`s dimensionality matches the one of a reference unit.
+
+    Parameters
+    ----------
+    da:
+        The data array that should be checked.
+    units_ref:
+        The reference unit
+
+    Raises
+    ------
+    pint.DimensionalityError:
+        The error is raised if the check fails
+
+    """
+    if units_ref is None:
+        return
+    if not isinstance(units_ref, pint.Unit):
+        units_ref = U_(units_ref)
+
+    units = da.attrs.get(UNITS_KEY, None)
+    if units is None and isinstance(da.data, pint.Quantity):
+        units = da.data.u
```

Comment:
Here I think it would be safer to first check the actual data for quantity and only check the attributes after (if we have a quantity that unit should take precedence over the attributes)
```suggestion
    units = da.pint.units
    if units is None:
        units = da.attrs.get(UNITS_KEY, None)
```